qt-bugs@ issue : 23458
bugs.kde.org number : 58040
applied: no
author: Stephan Binner <binner@kde.org>

Less flicker during changes in the tab bar.

No replacement for the disabled (identical) code parts given as I don't
understand the purpose of them and noticed no drawbacks when disabled.

Index: src/widgets/qtabbar.cpp
===================================================================
RCS file: /home/kde/qt-copy/src/widgets/qtabbar.cpp,v
retrieving revision 1.52
diff -u -3 -p -b -r1.52 qtabbar.cpp
--- src/widgets/qtabbar.cpp	23 Jun 2003 11:48:57 -0000	1.52
+++ src/widgets/qtabbar.cpp	1 Jul 2003 12:29:42 -0000
@@ -1290,7 +1290,7 @@ void QTab::setText( const QString& text 
 	    tb->d->a->insertItem( p, id );
 #endif
 	tb->layoutTabs();
-	tb->repaint();
+	tb->repaint( FALSE );
 
 #if defined(QT_ACCESSIBILITY_SUPPORT)
 	QAccessible::updateAccessibility( tb, tb->indexOf(id)+1, QAccessible::NameChanged );
Index: src/widgets/qtabwidget.cpp
===================================================================
RCS file: /home/kde/qt-copy/src/widgets/qtabwidget.cpp,v
retrieving revision 1.50
diff -u -3 -p -b -r1.50 qtabwidget.cpp
--- src/widgets/qtabwidget.cpp	23 Jun 2003 11:48:57 -0000	1.50
+++ src/widgets/qtabwidget.cpp	1 Jul 2003 12:29:43 -0000
@@ -380,7 +380,7 @@ void QTabWidget::changeTab( QWidget *w, 
         return;
     // this will update the accelerators
     t->setText( label );
-
+/*
     int ct = d->tabs->currentTab();
     bool block = d->tabs->signalsBlocked();
     d->tabs->blockSignals( TRUE );
@@ -389,7 +389,7 @@ void QTabWidget::changeTab( QWidget *w, 
 	d->tabs->setCurrentTab( ft->identifier() );
     d->tabs->setCurrentTab( ct );
     d->tabs->blockSignals( block );
-
+*/
     d->tabs->update();
     setUpLayout();
 }
@@ -414,7 +414,7 @@ void QTabWidget::changeTab( QWidget *w, 
     // this will update the accelerators
     t->iconset = new QIconSet( iconset );
     t->setText( label );
-
+/*
     int ct = d->tabs->currentTab();
     bool block = d->tabs->signalsBlocked();
     d->tabs->blockSignals( TRUE );
@@ -423,7 +423,7 @@ void QTabWidget::changeTab( QWidget *w, 
 	d->tabs->setCurrentTab( ft->identifier() );
     d->tabs->setCurrentTab( ct );
     d->tabs->blockSignals( block );
-
+*/
     d->tabs->update();
     setUpLayout();
 }
@@ -1041,7 +1041,7 @@ void QTabWidget::setTabIconSet( QWidget 
     t->iconset = new QIconSet( iconset );
 
     d->tabs->layoutTabs();
-
+/*
     int ct = d->tabs->currentTab();
     bool block = d->tabs->signalsBlocked();
     d->tabs->blockSignals( TRUE );
@@ -1050,7 +1050,7 @@ void QTabWidget::setTabIconSet( QWidget 
 	d->tabs->setCurrentTab( ft->identifier() );
     d->tabs->setCurrentTab( ct );
     d->tabs->blockSignals( block );
-
+*/
     d->tabs->update();
     setUpLayout();
 }
